app/src/main/res/layout/list_item_message_received.xml

```diff
@@ -25,22 +26,39 @@
         android:layout_marginLeft="@dimen/meshim_margin_small"
         android:layout_marginStart="@dimen/meshim_margin_small"
         android:background="@drawable/chat_bubble_rec"
-        android:padding="@dimen/meshim_margin_small"
         android:elevation="@dimen/elevation_card"
+        android:padding="@dimen/meshim_margin_small"
         android:textColor="@color/white"
-        app:layout_constraintStart_toEndOf="@id/image_message_profile"
         app:layout_constraintLeft_toRightOf="@+id/image_message_profile"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintWidth_default="wrap"
-        app:layout_constraintHorizontal_bias="0.0" />
+        app:layout_constraintStart_toEndOf="@id/image_message_profile"
+        app:layout_constraintWidth_default="wrap" />
 
     <TextView
         android:id="@+id/text_message_time_recieved"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginTop="@dimen/meshim_margin_mini"
         android:textSize="12sp"
         app:layout_constraintLeft_toLeftOf="@+id/text_message_body"
-        app:layout_constraintTop_toBottomOf="@+id/text_message_body" />
+        app:layout_constraintTop_toBottomOf="@+id/text_recieve_image" />
+
+    <TextView
+        android:id="@+id/text_file_recieved"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/meshim_margin_small"
+        android:layout_marginTop="@dimen/meshim_margin_small"
+        android:autoLink=""
+        app:layout_constraintStart_toEndOf="@+id/text_message_body"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <ImageView
+        android:id="@+id/text_recieve_image"
+        android:layout_width="200dp"
```

Comment:
Should be a constant, I think we have some "square" ones that might be re-usable.

Reply:
Same deal with the sent layout.